examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "BdxOtaSender.h"
+
+#include <core/CHIPError.h>
+#include <messaging/ExchangeContext.h>
+#include <protocols/bdx/BdxTransferSession.h>
+#include <support/BitFlags.h>
+
+#include <fstream>
+
+#include <string.h>
+
+using namespace chip::bdx;
+
+void BdxOtaSender::SetFilepath(const char * path, size_t length)
+{
+    if (path != nullptr)
+    {
+        strcpy(mFilepath, path);
```

Comment:
should the length be used?

`ChipMemString.h` can be used to ensure zero terminator and no memory overflow on mFilePath.

Reply:
I've been looking for something like `ChipMemString.h`! I'll see how I can integrate it